### PR TITLE
Infra 4366 time bug

### PR DESCRIPTION
**Pull request by elainearbaugh**
Panels can override the dashboard time settings, take this into account when creating a check.

Also set the query time range to the check time range in the auto_sync task

### CODE REVIEW

**Comment by elainearbaugh** [comment]
trigger build

File: cabot/metricsapp/forms/grafana_elastic.py [hunk 1]
```diff
@@ -68,3 +68,10 @@ def __init__(self, *args, **kwargs):
         self.fields['queries'].widget = forms.Textarea(attrs=dict(readonly='readonly',
                                                                   style='width:100%'))
         self.fields['queries'].help_text = None
+
+    def save(self):
+        """Adjust extended_bounds part of queries if the time_range is changed"""
+        model = super(GrafanaElasticsearchStatusCheckUpdateForm, self).save()
+        model.queries = json.dumps(adjust_time_range(json.loads(model.queries), model.time_range))
```
Comment (mahendra) [inline cabot/metricsapp/forms/grafana_elastic.py]: Isn't there a better place to do this? IIRC, Django had some hooks for cleaning up each field before saving. Can we use that.

Comment (elainearbaugh) [inline cabot/metricsapp/forms/grafana_elastic.py]: good call

File: cabot/metricsapp/signals.py [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import json
+from django.db.models.signals import pre_save
+from django.dispatch import receiver
+from cabot.metricsapp.api import adjust_time_range
+from cabot.metricsapp.models import ElasticsearchStatusCheck
+
+
+@receiver(pre_save, dispatch_uid="adjust_query_time_range", sender=ElasticsearchStatusCheck)
```
Comment (mahendra) [inline cabot/metricsapp/signals.py]: 👍

**Pull request merged**